### PR TITLE
Add support for osu!mania and osu!taiko scores

### PR DESCRIPTION
**Pull request by Rekkonnect**
Displays the count of misses, correctly displays the scores for an osu!mania or osu!taiko score.

This does affect the score overview in the replay score overview circle, overflowing osu!mania scores, but that's to be taken care of separately.

To be considered:
- Use multiple columns with spacing for each judgement type in order to avoid confusion

TODO:
- [ ] Add support for osu!catch scores (will need to decide on the judgement types that will be shown)

### CODE REVIEW

File: osu.Game/Overlays/BeatmapSet/Scores/DrawableTopScore.cs [hunk 1]
```diff
@@ -58,7 +58,21 @@ public OnlineScore Score
 
                 totalScore.Value = $@"{score.TotalScore:N0}";
                 accuracy.Value = $@"{score.Accuracy:P2}";
-                statistics.Value = $"{score.Statistics[HitResult.Great]}/{score.Statistics[HitResult.Good]}/{score.Statistics[HitResult.Meh]}";
+                if (score.Statistics.ContainsKey(HitResult.Perfect)) // Only osu!mania contains this judgement type currently
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/Scores/DrawableTopScore.cs]: this conditional section is pretty ugly

**Comment by peppy** [comment]
Things have changed a *lot* since this PR. I believe scores should be handled in a much nicer way already. @smoogipoo confirm that this PR can be closed?

**Comment by smoogipoo** [comment]
Probably safe to close, it's 2+ years out of date.

**Comment by smoogipoo** [comment]
Going forward there should be a function exposed from `Ruleset` that gives the relevant `HitResult`s and their namings for a ruleset.

Closing this as it's severely out of date and shouldn't be hardcoded like this anyway.

**Pull request closed without merging**